src/v/redpanda/admin_server.cc

```diff
@@ -4819,3 +4856,76 @@ admin_server::restart_service_handler(std::unique_ptr<ss::http::request> req) {
     co_await restart_redpanda_service(*service);
     co_return ss::json::json_return_type(ss::json::json_void());
 }
+
+debug_bundle_params make_debug_bundle_params(const ss::http::request& req) {
+    // NOTE: Here, we sanitize for ss::experimental::process (i.e., posix_spawn)
+    // instead of RPK
+    const std::regex bundle_regex("[a-zA-Z0-9\\-\\.]+");
+    auto maybe_sanitize_debug_bundle_param =
+      [&bundle_regex](ss::sstring p) -> std::optional<ss::sstring> {
+        if (p.empty()) {
+            return std::nullopt;
+        }
+
+        validate_utf8(p);
+        if (!std::regex_match(p.c_str(), bundle_regex)) {
+            throw ss::httpd::server_error_exception(fmt::format(
+              "Query param contains invalid characters: param {}", p));
+        }
+
+        return std::make_optional(p);
+    };
+
+    debug_bundle_params bundle_params;
+    bundle_params.logs_since = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("logs-since"));
+    bundle_params.logs_until = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("logs-until"));
+    bundle_params.logs_size_limit = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("logs-size-limit"));
+    bundle_params.metrics_interval = maybe_sanitize_debug_bundle_param(
+      req.get_query_param("metrics-interval"));
```

Comment:
mentioned in the other PR, it might create a nicer interface if the admin server does the verification and conversion to data types like size_t and time_point.

Reply:
> I believe we are expecting the same format as being passed to `rpk`. 

Yes, I implemented these as strings since they are quickly routed to RPK and that also has validation.

> Do we want to validate these just to have `rpk` do the same validation?

I agree that is is strange to do verification of the same params twice, is there another motivation to use more specific types?

Reply:
>I agree that is is strange to do verification of the same params twice, is there another motivation to use more specific types?

Yes, it's easier to test the debug bundle. I mean this is a minor thing, but it is good design to keep interfaces clean.

Reply:
> > I agree that is is strange to do verification of the same params twice, is there another motivation to use more specific types?
> 
> Yes, it's easier to test the debug bundle. I mean this is a minor thing, but it is good design to keep interfaces clean.

I can appreciate this reasoning.  I think the 'tricky' (and it's probably not very tricky) is verifying the date string in the same manner that `rpk` does its validation (I'm assuming the logs-size-limit and metrics-interal are just integers)

Reply:
> I can appreciate this reasoning. I think the 'tricky' (and it's probably not very tricky) is verifying the date string in the same manner that rpk does its validation (I'm assuming the logs-size-limit and metrics-interal are just integers)

I'm not suggesting that we don't do verification or rpk-specific stuff. I'm just saying that some of that can be done at the admin server layer. The debug bundle definitely needs to be aware that it is interacting with rpk, but the caller doesn't need to know how to format strings properly for rpk to use the debug bundle.